cmd/kubeadm/app/util/apiclient/wait_test.go

```diff
@@ -73,10 +84,330 @@ func TestGetControlPlaneComponents(t *testing.T) {
 
 	for _, tc := range testcases {
 		t.Run(tc.name, func(t *testing.T) {
+			// Call the function under test
 			actual := getControlPlaneComponents(tc.cfg)
+			// Compare the expected and actual results
 			if !reflect.DeepEqual(tc.expected, actual) {
 				t.Fatalf("expected result: %+v, got: %+v", tc.expected, actual)
 			}
 		})
 	}
 }
+
+// TestGetStaticPodSingleHash tests that getStaticPodSingleHash correctly retrieves the static pod hash from annotations.
+func TestGetStaticPodSingleHash(t *testing.T) {
+	// Initialize a fake clientset
+	client := clientfake.NewSimpleClientset()
+
+	nodeName := "node1"
+	component := "kube-apiserver"
+	podName := fmt.Sprintf("%s-%s", component, nodeName)
+	expectedHash := "abc123"
+
+	// Create annotations with the expected hash
+	annotations := map[string]string{
+		"kubernetes.io/config.hash": expectedHash,
+	}
+
+	// Create a pod with the annotations
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        podName,
+			Namespace:   metav1.NamespaceSystem,
+			Annotations: annotations,
+		},
+	}
+
+	// Create the pod in the fake clientset
+	_, err := client.CoreV1().Pods(metav1.NamespaceSystem).Create(context.Background(), pod, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Failed to create pod: %v", err)
+	}
+
+	// Call getStaticPodSingleHash and verify the result
+	hash, err := getStaticPodSingleHash(client, nodeName, component)
+	if err != nil {
+		t.Errorf("getStaticPodSingleHash returned error: %v", err)
+	}
+
+	if hash != expectedHash {
+		t.Errorf("Expected hash %s, got %s", expectedHash, hash)
+	}
+}
+
+// TestGetStaticPodSingleHash_PodNotFound tests that getStaticPodSingleHash returns an error when the pod is not found.
+func TestGetStaticPodSingleHash_PodNotFound(t *testing.T) {
```

Comment:
we use subtests with testcases instead of seperate functions that do a different test.
check how TestGetControlPlaneComponents is doing it.